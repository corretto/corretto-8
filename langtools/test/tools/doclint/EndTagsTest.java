/*
 * @test /nodynamiccopyright/
 * @bug 8006236
 * @summary doclint: structural issue hidden
 * @build DocLintTester
 * @run main DocLintTester -Xmsgs:-html EndTagsTest.java
 * @run main DocLintTester -ref EndTagsTest.out EndTagsTest.java
 */

/** */
public class EndTagsTest {
    /** <p>  <a name="a1"> text <img alt="image" src="image.png"> </a> </p> */
    public void valid_all() { }

    /** <p>  <a name="a2"> text <img alt="image" src="image.png"> </a> */
    public void valid_omit_optional_close() { }

    /** </a> */
    public void invalid_missing_start() { }

    /** <p> </a> */
    public void invalid_missing_start_2() { }

    /** <p> text </p> </a> */
    public void invalid_missing_start_3() { }

    /** <img alt="image" src="image.png"> </img> */
    public void invalid_end() { }

    /** <invalid> </invalid> */
    public void unknown_start_end() { }

    /** <invalid> */
    public void unknown_start() { }

    /** </invalid> */
    public void unknown_end() { }
}

