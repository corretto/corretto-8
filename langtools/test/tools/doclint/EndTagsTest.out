EndTagsTest.java:18: error: unexpected end tag: </a>
    /** </a> */
        ^
EndTagsTest.java:21: error: unexpected end tag: </a>
    /** <p> </a> */
            ^
EndTagsTest.java:24: error: unexpected end tag: </a>
    /** <p> text </p> </a> */
                      ^
EndTagsTest.java:27: error: invalid end tag: </img>
    /** <img alt="image" src="image.png"> </img> */
                                          ^
EndTagsTest.java:30: error: unknown tag: invalid
    /** <invalid> </invalid> */
        ^
EndTagsTest.java:30: error: unknown tag: invalid
    /** <invalid> </invalid> */
                  ^
EndTagsTest.java:33: error: unknown tag: invalid
    /** <invalid> */
        ^
EndTagsTest.java:36: error: unknown tag: invalid
    /** </invalid> */
        ^
8 errors
